hack/update-template/main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+// Template is the k8s template type. to keep dependencies minimal, k8s api is not pulled in
+type Template struct {
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+	Parameters []struct {
+		Name     string `yaml:"name"`
+		Value    string `yaml:"value"`
+		Required bool   `yaml:"required"`
+	}
+	Metadata Metadata      `yaml:"metadata"`
+	Objects  []interface{} `yaml:"objects"`
+}
+
+// Metadata is the k8s metadata type. to keep dependencies minimal, k8s api is not pulled in
+type Metadata struct {
+	Name string `yaml:"name"`
+}
+
+func main() {
+	ex, getExecutableErr := os.Executable()
+	if getExecutableErr != nil {
+		panic(fmt.Errorf("could not get the Executable: %w", getExecutableErr))
+	}
+
+	workingDirectory := filepath.Dir(ex)
+	fmt.Printf("workingDirectory :: %s\n", workingDirectory)
+
+	var filesNames []string
+	relativeDeployFolder := filepath.Join(workingDirectory, "../../deploy/")
+
+	// get all the yaml files in the deploy folder
+	walkErr := filepath.Walk(relativeDeployFolder, func(path string, info os.FileInfo, err error) error {
+		if !strings.HasSuffix(path, "yaml") {
+			return nil
+		}
+		filesNames = append(filesNames, path)
+		return nil
+	})
+
+	if walkErr != nil {
+		panic(fmt.Errorf("error while searching files in '%s': %w", relativeDeployFolder, walkErr))
+	}
+
+	// this holds all of the info that is static in the template. update this as needed
+	saasTemplateFile := Template{
+		APIVersion: "v1",
+		Kind:       "Template",
+		Metadata: Metadata{
+			Name: "configuration-anomaly-detection-template",
+		},
+	}
+
+	// iterate through each of the files
+	for _, fileName := range filesNames {
+		// read the file
+		fileName = filepath.Clean(fileName)
+		fileAsBytes, readErr := ioutil.ReadFile(fileName) //#nosec G304 -- This is the best I can do :/
+
+		if readErr != nil {
+			panic(fmt.Errorf("could not read file '%s': %w", fileName, readErr))
+		}
+
+		fileAsMap := make(map[interface{}]interface{})
+
+		// marshal into a generic yaml blob
+		marshalErr := yaml.Unmarshal(fileAsBytes, &fileAsMap)
+
+		if marshalErr != nil {
+			panic(fmt.Errorf("could not marshal file '%s': %w", fileName, marshalErr))
+		}
+
+		// append to the objects section in the saas file
+		saasTemplateFile.Objects = append(saasTemplateFile.Objects, fileAsMap)
+	}
+
+	// marshal back into bytes
+	saasTemplateFileAsBytes, marshalBackErr := yaml.Marshal(&saasTemplateFile)
+	if marshalBackErr != nil {
+		panic(fmt.Errorf("could not marshal the bigMapping back: %v", marshalBackErr))
+	}
+	// print to stdout just so we know something happened and the code isn't broken
+	fmt.Printf("---\n%s\n---\n", string(saasTemplateFileAsBytes))
+	outputFile := filepath.Join(workingDirectory, "../configuration-anomaly-detection-template.Template.yaml")
```

Comment:
The target directory for the created file should be openshift/template.yaml (that's what all projects are doing). We can change the path in the app-sre deployment spec, but then it will be different to all other projects :thinking: 

I  don't have a clear opinion about this.

Reply:
Give the path and I will do it :)

Reply:
See my other PR for the path.. it's really just <project root>/openshift/template.yaml as specified here: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/38300/diffs#51d44de8da2934b1e8168e02e247073fe65f428f_0_41

Line 41